Code/Framework/AzToolsFramework/Tests/PaintBrush/PaintBrushPaintSettingsTests.cpp

```diff
@@ -0,0 +1,839 @@
+                        AzToolsFramework::PaintBrushNotifications::SmoothFn& smoothFn)
+                    {
+                        for (float baseValue = 0.0f; baseValue <= 1.0f; baseValue += 0.25f)
+                        {
+                            for (float newValue = 0.0f; newValue <= 1.0f; newValue += 0.25f)
+                            {
+                                // Create a 3x3 set of kernelValues all with newValue. The mean of this will be newValue,
+                                // so the output of smoothFn should be the same as blendFn for the same combinations of values.
+                                AZStd::vector<float> kernelValues(9, newValue);
+
+                                for (float opacity = 0.0f; opacity < 1.0f; opacity += 0.25f)
+                                {
+                                    float expectedValue = AZStd::clamp(verifyFn(baseValue, newValue, opacity), 0.0f, 1.0f);
+                                    EXPECT_NEAR(smoothFn(baseValue, kernelValues, opacity), expectedValue, 0.001f);
+                                }
+                            }
+                        }
+                    });
+
+            paintBrush.BeginBrushStroke(m_settings);
+            paintBrush.SmoothToLocation(TestBrushCenter, m_settings);
+            paintBrush.EndBrushStroke();
+
+            paintBrush.EndPaintMode();
+        }
+    };
+
+    TEST_F(PaintBrushPaintSettingsTestFixture, ZeroOpacityBrushSettingCausesNoNotifications)
+    {
+        // If the opacity is zero (transparent), OnPaint/OnSmooth will never get called because no points can get modified.
+
+        m_settings.SetColor(AZ::Color(0.0f));
+        TestZeroNotificationsForPaintAndSmooth();
+    }
+
+    TEST_F(PaintBrushPaintSettingsTestFixture, SizeBrushSettingAffectsPaintBrush)
+    {
+        // The PaintBrush 'Size' setting should affect the overall size of the paint brush circle that's being used to paint/smooth.
+
+        AzToolsFramework::PaintBrush paintBrush(EntityComponentIdPair);
+        ::testing::NiceMock<MockPaintBrushNotificationBusHandler> mockHandler(EntityComponentIdPair);
+
+        // Loop through a series of different brush radius sizes.
+        for (auto& brushRadiusSize : {0.5f, 1.0f, 5.0f, 10.0f, 20.0f})
+        {
+            m_settings.SetSize(brushRadiusSize * 2.0f);
+
+            ValidationFn validateFn = [=](const AZ::Aabb& dirtyArea,
+                                  AzToolsFramework::PaintBrushNotifications::ValueLookupFn& valueLookupFn)
+            {
+                // The dirtyArea AABB should change size based on the current brush radius size that we're using.
+                EXPECT_THAT(dirtyArea, IsClose(AZ::Aabb::CreateCenterRadius(TestBrushCenter2d, brushRadiusSize)));
+
+                // Create a 3x3 square grid of points. Because our brush is a circle, we expect only the points in along a + to be
```

Comment:
The wording of `in along a +` confused me a bit, maybe just `along a +` ?

Reply:
Whoops, typo, thx!